### DISCUSSION TITLE
Github client

### DISCUSSION THREAD

**Discussion by atompie** [Tracardi/tracardi#715] [General]
We would like to make Tracardi useful for the open-source community: Now tracardi can handle a lot of operations for e-commerce  customers. But we would like it to be used also as a open-source github automation tool. What I mean by this.  

Tracardi can collect difference events. I can collect event from the git-hub  webhook. It can be, for example, a first comment on the issue, or a a PR , etc. We can collect this already, we can process it (e.g check if it was first comment) but we can not send back information to git-hub. For this we need a client that will use GitHub API and do some limited (just for start) operations:

I am thinking about:

- make a comment on an issue
- assign an issue to the contributor
- etc.

The purpose is to make Tracardi an automation tool for interacting with the contributors. For example if a contributor forks Tracardi (or any repo) it would be nice to send him a tutorial on now to use the system and the welcome message. If, for example, there is a comment and the admin decides that all the comments should be check by AI for the spam, or bad language (we can do it with tracardi) then we could tag (label) this comment as inappropriate if the language is bad.

So we need a set of actions that we could do in the github to help maintainers and contributors to automate some tasks. 

If you have any interesting use-case in mind please share below.

**Comment by detronetdip** [comment]
hey you want to make it as a software(eg: with electron) or some web based application?

**Comment by atompie** [comment]
We need it as python package. It will become a part of Tracardi workflow.
